nupic/research/frameworks/pytorch/imagenet/mixins/vary_batch_size.py

```diff
@@ -0,0 +1,104 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from copy import deepcopy
+
+
+class VaryBatchSize(object):
+    """
+    This mixin enables loading the training data with varying batch-sizes from
+    epoch to epoch.
+
+    :param config:
+        - batch_sizes: list of batch sizes; the last one will be used for the remainder
+                       of training once all have been exhausted, going in order
+        - batch_size: Can be given as a trivial case where the batch size doesn't vary;
```

Comment:
Can't that be a single argument, and `batch_size` accept a list as well? I find it a lot cleaner, but opinion may vary.

Reply:
I do want it to be clear that there are multiple batch-sizes. I've removed `batch_size` as an argument and now allow just `batch_sizes` for less ambiguity.